### ISSUE TITLE
ionic cordova build ios fails - Error code 65

### ISSUE THREAD

**Issue by alexnu** [chemerisuk/cordova-plugin-firebase-analytics#39]
I tried building my app in iOS for the first time after having added cordova-plugin-firebase-analytics, and I am getting the following error:

```
[ERROR] Cordova encountered an error.
        You may get more insight by running the Cordova command above directly.
        
[ERROR] An error occurred while running cordova build ios --release (exit code 1):
        
        (truncated) ...
        e.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c
        -arch i386 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=c99
        -fobjc-arc -fmodules -gmodules -fmodules-cache-path=/Users/nafas/Library/Developer/Xcode/DerivedData/ModuleCache
        -fmodules-prune-interval=86400 -fmodules-prune-after=345600
        -fbuild-session-file=/Users/nafas/Library/Developer/Xcode/DerivedData/ModuleCache/Session.modulevalidation
        -fmodules-validate-once-per-build-session -Wno-trigraphs -fpascal-strings -Os -Wno-missing-field-initializers
        -Wno-missing-prototypes -Wno-implicit-atomic-properties -Wno-arc-repeated-use-of-weak -Wduplicate-method-match
        -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter
        -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow
        -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion
        -Wenum-conversion -Wno-shorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match
        -Wundeclared-selector -Wno-deprecated-implementations -isysroot
        /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.3.sdk
        -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -mios-simulator-version-min=9.0 -g
        -fvisibility=hidden -Wno-sign-conversion -Wno-infinite-recursion -fobjc-abi-version=2 -fobjc-legacy-dispatch
        -iquote
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Updoc-generated-files.hmap
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Updoc-own-target-headers.hmap
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Updoc-all-target-headers.hmap
        -iquote
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Updoc-project-headers.hmap
        -I/Users/nafas/updoc/app/platforms/ios/build/emulator/include
        -I/Users/nafas/updoc/app/platforms/ios/build/emulator/usr/local/lib/include
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/UninstalledProducts/include
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/UninstalledProducts/iphonesimulator/include
        -I/Users/nafas/updoc/app/platforms/ios/build/emulator
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/DerivedSources/i386
        -I/Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/DerivedSources
        -F/Users/nafas/updoc/app/platforms/ios/build/emulator -include
        /Users/nafas/updoc/app/platforms/ios/build/sharedpch/Updoc-Prefix-hdaupibbgcnigebntimpqhbbwlsk/Updoc-Prefix.pch
        -MMD -MT dependencies -MF
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Objects-normal/i386/CDVInAppBrowser.d
        --serialize-diagnostics
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Objects-normal/i386/CDVInAppBrowser.dia
        -c /Users/nafas/updoc/app/platforms/ios/Updoc/Plugins/cordova-plugin-inappbrowser/CDVInAppBrowser.m -o
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Objects-normal/i386/CDVInAppBrowser.o
        
        ** BUILD FAILED **
        
        
        The following build commands failed:
        CompileC
        /Users/nafas/Library/Developer/Xcode/DerivedData/Updoc-fklblfhvmnlqgpfrqmiiumwouyfz/Build/Intermediates/Updoc.build/Release-iphonesimulator/Updoc.build/Objects-normal/i386/FirebaseAnalyticsPlugin.o
        Updoc/Plugins/cordova-plugin-firebase-analytics/FirebaseAnalyticsPlugin.m normal i386 objective-c
        com.apple.compilers.llvm.clang.1_0.compiler
        (1 failure)
        Error: Error code 65 for command: xcodebuild with args:
        -xcconfig,/Users/nafas/updoc/app/platforms/ios/cordova/build-release.xcconfig,-workspace,Updoc.xcworkspace,-scheme,Updoc,-configuration,Release,-sdk,iphonesimulator,-destination,platform=iOS
        Simulator,name=iPhone
        SE,build,CONFIGURATION_BUILD_DIR=/Users/nafas/updoc/app/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/nafas/updoc/app/platforms/ios/build/sharedpch
```

**Comment by anhiliate** [comment]
same goes for me, any solutions?

**Comment by saoron** [comment]
In my case I forgot to delete **cordova-plugin-firebase** and got error 65.

Try running

`cordova rm cordova-plugin-firebase`

**Comment by alexnu** [comment]
I don't have **cordova-plugin-firebase** installed, only firebase 4.1.3 on my package.json but I don't think that causes the problem.

**Comment by lordgreg** [comment]
Exactly the same issue. 

Currently no idea what more to do.

**Comment by dheerajb** [comment]
I am also facing the same issue. It was all fine suddenly this error started showing.

**Comment by Gradix93** [comment]
I had same issue mostly when i tried to install the plugin i had an error **Failed to install 'cordova-plugin-firebase-analytics':undefined
Error: CocoaPods was not found. Please install version 1.0.1 or greater from https://cocoapods.org/**.

My solution is to follow these steps:
1) Force to uninstall plugin "**cordova plugin remove cordova-plugin-firebase-analytics --save**" (until you get this error "**Error: Plugin "cordova-plugin-firebase-analytics" is not present in the project. See `cordova plugin list`.**"

2) So you must update your cocoapods following these steps "**sudo gem install cocoapods**" (to get last version of cocoapods) and "**pod setup**" (for sync new version into your OS)

Then, now you can reinstall plugin "**cordova plugin add cordova-plugin-firebase-analytics --save**".

The build will be success

Note: my supposition of this plugin that you can build only with Xcode 8+ (because i have 7.2 and i had that issue)

**Comment by vadimwe** [comment]
No luck. Just compile error

**Comment by brunolm** [comment]
iOS

- Open `platforms/ios` on XCode
- Find & Replace `io.ionic.starter` in all files for a unique identifier
- Click the project to open settings
- Signing > Select a team
- Go to your device Settings > General > DeviceManagement
  - Trust your account/team
- `ionic cordova run ios --device --livereload`

**Comment by pravs2011** [comment]
my ionic iOS build is failing 

(node:13834) UnhandledPromiseRejectionWarning: Unhandled promise rejection.

**Comment by bhumin3i** [comment]
@pravs2011  same problem
sample log : 

```
Ld build/emulator/RTI.app/RTI normal x86_64
    cd /Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=9.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/bhumin/Desktop/gitREPO/rtidev/node_modules/.bin:/Users/bhumin/Library/Android/sdk//tools:/Users/bhumin/Library/Android/sdk//platform-tools:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk -L/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/emulator -L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/GoogleToolboxForMac -L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/Protobuf -L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/nanopb -F/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/emulator -F/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/Pods/FirebaseAnalytics/Frameworks -F/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/Pods/FirebaseCore/Frameworks -F/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/Pods/FirebaseInstanceID/Frameworks -F/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/Pods/FirebaseMessaging/Frameworks -FRTI/Plugins/com.googlemaps.ios -FRTI/Plugins/cordova-plugin-ms-adal -filelist /Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Intermediates.noindex/RTI.build/Debug-iphonesimulator/RTI.build/Objects-normal/x86_64/RTI.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=9.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Intermediates.noindex/RTI.build/Debug-iphonesimulator/RTI.build/Objects-normal/x86_64/RTI_lto.o -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -ObjC -ObjC -lGoogleToolboxForMac -lProtobuf -lc++ -lnanopb -lsqlite3 -lz -framework AddressBook -framework FirebaseAnalytics -framework FirebaseCore -framework FirebaseCoreDiagnostics -framework FirebaseInstanceID -framework FirebaseMessaging -framework FirebaseNanoPB -framework StoreKit -framework SystemConfiguration -ObjC -lGoogleToolboxForMac -lProtobuf -lc++ -lnanopb -lsqlite3 -lz -framework AddressBook -framework FirebaseAnalytics -framework FirebaseCore -framework FirebaseCoreDiagnostics -framework FirebaseInstanceID -framework FirebaseMessaging -framework FirebaseNanoPB -framework StoreKit -framework SystemConfiguration -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Intermediates.noindex/RTI.build/Debug-iphonesimulator/RTI.build/RTI.app.xcent /Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/emulator/libCordova.a -framework CoreLocation -framework CoreTelephony -framework CoreImage -lz -framework QuartzCore -framework SystemConfiguration -weak_framework WebKit -framework ADAL -framework UserNotifications -liconv -framework AVFoundation -framework CoreVideo -framework AudioToolbox -framework Accelerate -framework CoreData -framework CoreText -framework GLKit -framework ImageIO -lc++ -framework OpenGLES -framework UIKit -framework GoogleMapsBase -framework GoogleMaps -framework GoogleMapsCore -Xlinker -dependency_info -Xlinker /Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Intermediates.noindex/RTI.build/Debug-iphonesimulator/RTI.build/Objects-normal/x86_64/RTI_dependency_info.dat -o /Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/emulator/RTI.app/RTI
ld: warning: directory not found for option '-L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/GoogleToolboxForMac'
ld: warning: directory not found for option '-L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/Protobuf'
ld: warning: directory not found for option '-L/Users/bhumin/Library/Developer/Xcode/DerivedData/RTI-aqtvytpqvcxbkaepcfkfyqitbgqv/Build/Products/Debug-iphonesimulator/nanopb'
ld: library not found for -lGoogleToolboxForMac
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **


The following build commands failed:
	Ld build/emulator/RTI.app/RTI normal x86_64
(1 failure)
(node:72758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error code 65 for command: xcodebuild with args: -xcconfig,/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/cordova/build-debug.xcconfig,-workspace,RTI.xcworkspace,-scheme,RTI,-configuration,Debug,-sdk,iphonesimulator,-destination,platform=iOS Simulator,name=iPhone X,build,CONFIGURATION_BUILD_DIR=/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/bhumin/Desktop/gitREPO/rtidev/platforms/ios/build/sharedpch
(node:72758) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by ilyalukyanov** [comment]
@bhumin3i your error looks different to the one the issue is about, although error code is the same. However, I've been getting exactly the same output as yours and found a workaround.

For some reason just re-adding the plugin fixes it for me. I added it to my npm postinstall script:
```
"postinstall": "ionic cordova prepare && ionic cordova plugin remove cordova-plugin-firebase-analytics && ionic cordova plugin add cordova-plugin-firebase-analytics@0.13.0 --variable FIREBASE_VERSION=\"11.6.+\"",
``` 
Hope this helps.

**Comment by dimitri320** [comment]
So, to avoid building Error, just make sure you CLOSE the Xcode window before you add the plugin, and open it only after building the app. This way all should work just fine.

Another issue comes up after though: The App terminates with the following error: 

`*** Terminating app due to uncaught exception 'com.firebase.core', reason: 'Default app has already been configured.'`

**Comment by hptumovil** [comment]
Same problem, POD install doesn't fix it. any help?

* BUILD FAILED **


The following build commands failed:
        CompileC /var/root/Library/Developer/Xcode/DerivedData/HPTUMovil-brdzlsijtnmxcvdagzhuqcajuvwg/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FirebaseCore.build/Objects-normal/x86_64/FIRNetwork.o FirebaseCore/Firebase/Core/FIRNetwork.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
(node:5795) UnhandledPromiseRejectionWarning: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/hptumovil/Documents/HPTUMovil/platforms/ios/cordova/build-debug.xcconfig,-workspace,HPTUMovil.xcworkspace,-scheme,HPTUMovil,-configuration,Debug,-sdk,iphonesimulator,-destination,platform=iOS Simulator,name=iPhone X,build,CONFIGURATION_BUILD_DIR=/Users/hptumovil/Documents/HPTUMovil/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/hptumovil/Documents/HPTUMovil/platforms/ios/build/sharedpch
(node:5795) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:5795) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by mohamedsharaf** [comment]
i confirm that this error only in command line when u open the project in xcode it runs without error

**Comment by muralijai** [comment]
For my recent App
I am seeing a build error when using command line build option
> sudo ionic cordova build ios --release
However, when I open my-proj.xworkspace and build it goes through fine but App crashes as expected.

For my previous App in May I was seeing this error in build option and I found that comment by
"Gradix93 commented on Oct 27, 2017" in above of this chain to be of perfect help.

However there is a sequence of steps one should maintain to go through successfully. I will post that sequence after I run it through my recent app. This is particularly after including firebase analytics in the App. cocoapads is the key to be in place. more soon.

**Comment by mohamedsharaf** [comment]
Gradix93 steps not working with me 

i update pods
 pod --version
1.5.3

**Comment by muralijai** [comment]
After several (I mean several several) rounds of experimenting, reading and testing here are the steps I have found to clearly get out of this issue. It worked for 2 iOS Apps have I applied to. Hope it helps all.

if you get this below message when you try to install firebase-analytics in MAC, here is how you go about it.
--------------------------------
Failed to install 'cordova-plugin-firebase-analytics': undefined

Failed to restore plugin "cordova-plugin-firebase-analytics" from config.xml. You might need to try adding it again. Error: /Library/Ruby/Gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:439:in `help!': [!] You cannot run CocoaPods as root. (CLAide::Help)
--------------------------------
I had to set the permission for the directory for everyone to read/write using getInfo menu item in MAC for that directory. This was because I don't have all rights to files.
--------------------------------
$ cordova plugin remove cordova-plugin-firebase-analytics --save
Uninstalling cordova-plugin-firebase-analytics from ios

[!] The Podfile does not contain any dependencies.

Removing "cordova-plugin-firebase-analytics"
Removing plugin cordova-plugin-firebase-analytics from config.xml file...
Removing cordova-plugin-firebase-analytics from package.json
--------------------------------
$ sudo gem install cocoapods
Successfully installed cocoapods-1.5.3
Parsing documentation for cocoapods-1.5.3
Done installing documentation for cocoapods after 4 seconds
1 gem installed
--------------------------------
$ pod setup
Setting up CocoaPods master repo
  $ /usr/local/bin/git -C /Users/general/.cocoapods/repos/master fetch origin --progress
  remote: Counting objects: 12879, done.
  remote: Compressing objects: 100% (194/194), done.
  remote: Total 12879 (delta 5221), reused 5165 (delta 5165), pack-reused 7507
  Receiving objects: 100% (12879/12879), 1.51 MiB | 321.00 KiB/s, done.
  Resolving deltas: 100% (8693/8693), completed with 1801 local objects.
  From https://github.com/CocoaPods/Specs
     76a31b08e7c..7f76a16af16  master     -> origin/master
  $ /usr/local/bin/git -C /Users/general/.cocoapods/repos/master rev-parse --abbrev-ref HEAD
  master
  $ /usr/local/bin/git -C /Users/general/.cocoapods/repos/master reset --hard origin/master
  Checking out files: 100% (1348/1348), done.
  HEAD is now at 7f76a16af16 [Add] LSiOSTableView 0.2.1
Setup completed
--------------------------------
I had to set the permission for the directory for everyone to read/write using getInfo menu item in MAC for that directory. This was because I don't have all rights to files.
--------------------------------
$ vi platform/ios/Podfile
ensure that in the Podfile the below shows
		pod 'Firebase/Core', '~> 4.1'
--------------------------------
$ pod install
Analyzing dependencies
Downloading dependencies
Installing Firebase (4.13.0)
Installing FirebaseAnalytics (4.2.0)
Installing FirebaseCore (4.0.20)
Installing FirebaseInstanceID (2.0.10)
Installing GoogleToolboxForMac (2.1.4)
Installing nanopb (0.3.8)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There is 1 dependency from the Podfile and 6 total pods installed.
--------------------------------
I had to set the permission for the directory for everyone to read/write using getInfo menu item in MAC for that directory. This was because I don't have all rights to files.
--------------------------------
Note the 0.13.0 version of firebase-analytics used here

$ cordova plugin add cordova-plugin-firebase-analytics@0.13.0 --save
Installing "cordova-plugin-firebase-analytics" for ios
Adding cordova-plugin-firebase-analytics to package.json
Saved plugin info for "cordova-plugin-firebase-analytics" to config.xml
--------------------------------
I had to set the permission for the directory for everyone to read/write using getInfo menu item in MAC for that directory. This was because I don't have all rights to files.
--------------------------------
$ sudo ionic cordova build ios --release
** BUILD SUCCEEDED **
--------------------------------
opened xcode using xcworkspace
set device to generic ios device
Shift+command+k - clean ok
command+b - build ok
connected iphone to MAC
xcode automatically sets device to connected iphone
Shift+command+k - clean ok
command+b - build ok
run app on device iphone
crashed because the GoogleService-Info.plist was not found
GoogleService-Info.plist file has to be in platform/ios folder
manually copied GoogleService-Info.plist file to platform/ios folder
Then added the GoogleService-Info.plist file using xcode project
run app on device iphone
worked fine this time
--------------------------------
DISCONNECT iPHONE before running below command
$ sudo ionic cordova build ios --release
** BUILD SUCCEEDED **

**Comment by yapching** [comment]
> Same problem, POD install doesn't fix it. any help?
> 
> * BUILD FAILED **
> 
> The following build commands failed:
> CompileC /var/root/Library/Developer/Xcode/DerivedData/HPTUMovil-brdzlsijtnmxcvdagzhuqcajuvwg/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FirebaseCore.build/Objects-normal/x86_64/FIRNetwork.o FirebaseCore/Firebase/Core/FIRNetwork.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
> (1 failure)
> (node:5795) UnhandledPromiseRejectionWarning: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/hptumovil/Documents/HPTUMovil/platforms/ios/cordova/build-debug.xcconfig,-workspace,HPTUMovil.xcworkspace,-scheme,HPTUMovil,-configuration,Debug,-sdk,iphonesimulator,-destination,platform=iOS Simulator,name=iPhone X,build,CONFIGURATION_BUILD_DIR=/Users/hptumovil/Documents/HPTUMovil/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/hptumovil/Documents/HPTUMovil/platforms/ios/build/sharedpch
> (node:5795) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
> (node:5795) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

$ sudo ionic cordova build ios --release

**Comment by hvaughan3** [comment]
@yapching The build will indeed fail but if you ignore that and go open the .xcworkspace file, the app actually does run fine.

**Comment by neilbo** [comment]
> I had same issue mostly when i tried to install the plugin i had an error **Failed to install 'cordova-plugin-firebase-analytics':undefined Error: CocoaPods was not found. Please install version 1.0.1 or greater from https://cocoapods.org/**.
> 
> My solution is to follow these steps:
> 
> 1. Force to uninstall plugin "**cordova plugin remove cordova-plugin-firebase-analytics --save**" (until you get this error "**Error: Plugin "cordova-plugin-firebase-analytics" is not present in the project. See `cordova plugin list`.**"
> 2. So you must update your cocoapods following these steps "**sudo gem install cocoapods**" (to get last version of cocoapods) and "**pod setup**" (for sync new version into your OS)
> 
> Then, now you can reinstall plugin "**cordova plugin add cordova-plugin-firebase-analytics --save**".
> 
> The build will be success
> 
> Note: my supposition of this plugin that you can build only with Xcode 8+ (because i have 7.2 and i had that issue)

**Should note that `pod setup` takes a very long time**
similar to this [stackoverflow issue](https://stackoverflow.com/questions/41079014/ionic-add-remove-phonegap-push-plugin-cocoapods-was-not-found)

**Comment by manel00** [comment]
Hello everyone, still not working guys? I have a similar error...

`** BUILD FAILED **


The following build commands failed:
        CompileC /Users/manel/Library/Developer/Xcode/DerivedData/TopStockAlerts-gymujfxkawqxonepqjfijwueghha/Build/Intermediates.noindex/TopStockAlerts.build/Debug-iphonesimulator/TopStockAlerts.build/Objects-normal/x86_64/FirebaseAnalyticsPlugin.o /Users/manel/Documents/ionic_projects/topstock/platforms/ios/TopStockAlerts/Plugins/cordova-plugin-firebase-analytics/FirebaseAnalyticsPlugin.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
        CompileC /Users/manel/Library/Developer/Xcode/DerivedData/TopStockAlerts-gymujfxkawqxonepqjfijwueghha/Build/Intermediates.noindex/TopStockAlerts.build/Debug-iphonesimulator/TopStockAlerts.build/Objects-normal/x86_64/AppDelegate+FCMPlugin.o /Users/manel/Documents/ionic_projects/topstock/platforms/ios/TopStockAlerts/Plugins/cordova-plugin-fcm-with-dependecy-updated/AppDelegate+FCMPlugin.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(2 failures)
xcodebuild: Command failed with exit code 65
[ERROR] An error occurred while running subprocess cordova.`

**Comment by nurzarifah** [comment]
I am facing quite similar with the problems. I am using ionic 3. I updated the macOS & Xcode 3 days before and I am not face any problem with that until I installed some plugin and cannot build. I achieved to solve this by using backup ios folder instead of current folder. 

Which is, I put current ios folder into another place and replace with the successful ios folder before (without the plugin) and try to build again. 

Hope it helps for some people.
